python/tk_desktop/desktop_engine_site_implementation.py

```diff
@@ -48,6 +49,78 @@ def destroy_engine(self):
         if self.msg_server is not None:
             self.msg_server.close()
 
+
+    ###########################################################################
+    # panel support (displayed as tabs)
+    def _init_tabs(self):
+        """
+        Create desktop tabs from the registered application panels, following
+        their specified relative positioning.
+        """
+        # make the "Apps" tab appear as a fake panel to fit in the sorting
+        # logic
+        apps_tab = {"callback":   self.desktop_window._register_apps_tab,
+                    "properties": {"position": 0}}
+
+        panels = [apps_tab] + list(self._engine.panels.itervalues())
+
+        def panel_position(panel):
+            """ Returns the relative position of a panel """
+            # sane checking on the expected structure
+            assert("properties" in panel, "panel should have properties set")
+            assert("position" in panel["properties"],
+                   "panel should have a position set")
+            return panel["properties"]["position"]
+
+        ordered_panels = sorted(panels, key=panel_position)
```

Comment:
I don't think this logic gets the outcome we talked about.  We are going for any apps with position set will come first in position order, and then the rest will show up in whatever order the register gets called, right?

If something is registered with position 1000, it would probably come last with this logic.

Seems like keeping position optional in the dictionary and defining a cmp that just orders according to the above criteria would give the right ordering.